Mathlib/Geometry/Euclidean/Altitude.lean

```diff
@@ -170,6 +171,101 @@ def height {n : ℕ} [NeZero n] (s : Simplex ℝ P n) (i : Fin (n + 1)) : ℝ :=
 lemma height_pos {n : ℕ} [NeZero n] (s : Simplex ℝ P n) (i : Fin (n + 1)) : 0 < s.height i := by
   simp [height]
 
+open scoped RealInnerProductSpace
+
+-- TODO: move
+private theorem _root_.Fin.exists_ne_of_ne_of_two_lt
+    {n : ℕ} {i j : Fin n} (h : i ≠ j) (h : 2 < n) : ∃ k, k ≠ i ∧ k ≠ j := by
+  have : NeZero n := ⟨by linarith⟩
+  rcases i with ⟨i, hi⟩
+  rcases j with ⟨j, hj⟩
+  simp_rw [← Fin.val_ne_iff]
+  by_cases h0 : 0 ≠ i ∧ 0 ≠ j
+  · exact ⟨0, h0⟩
+  · by_cases h1 : 1 ≠ i ∧ 1 ≠ j
+    · exact ⟨⟨1, by omega⟩, h1⟩
+    · refine ⟨⟨2, by omega⟩, ?_⟩
+      dsimp only
+      omega
+
+/-- The inner product of an edge from `j` to `i` and the vector from the foot of `i` to `i`
+is the square of the height. -/
+lemma inner_vsub_vsub_altitudeFoot_eq_height_sq
+    {n : ℕ} [NeZero n] (s : Simplex ℝ P n) {i j : Fin (n + 1)} (h : i ≠ j)  :
+    ⟪s.points i -ᵥ s.points j, s.points i -ᵥ s.altitudeFoot i⟫ = s.height i ^ 2 := by
+  rw [height, inner_vsub_vsub_left_eq_dist_sq_right_iff, altitudeFoot]
+  refine Submodule.inner_right_of_mem_orthogonal
+        (K := vectorSpan ℝ (Set.range (s.faceOpposite i).points))
+        (vsub_mem_vectorSpan_of_mem_affineSpan_of_mem_affineSpan
+          (SetLike.coe_mem _)
+          (mem_affineSpan ℝ ?_)) ?_
+  · simp only [range_faceOpposite_points, ne_eq, Set.mem_image, Set.mem_setOf_eq]
+    refine ⟨j, h.symm, rfl⟩
+  · rw [← direction_affineSpan]
+    exact vsub_orthogonalProjection_mem_direction_orthogonal _ _
+
+/-- The angle between two distinct faces can never reach 0 or π. -/
+lemma abs_inner_height_vsub_altitudeFoot_div_lt_one
+    {n : ℕ} [NeZero n] (s : Simplex ℝ P n) {i j : Fin (n + 1)} (hij : i ≠ j) (hn : 1 < n) :
+    |⟪s.points i -ᵥ s.altitudeFoot i, s.points j -ᵥ s.altitudeFoot j⟫
+            / (s.height i * s.height j)| < 1 := by
```

Comment:
@jsm28, can you think of a better docstring? This is of course describing the statement of a downstream lemma...

Reply:
Well, referring to the angle between two altitudes would be more precise. But to describe the lemma accurately would be more like "The inner product of two altitudes has absolute value less than the product of their lengths (equivalently, neither vector is a multiple of the other; the angle between them is not 0 or π).".